polygon/stream.go

```diff
@@ -238,14 +242,22 @@ func openSocket() *websocket.Conn {
 	if !ok {
 		polygonStreamEndpoint = "wss://alpaca.socket.polygon.io/stocks"
 	}
-	c, _, err := websocket.DefaultDialer.Dial(polygonStreamEndpoint, nil)
-	if err != nil {
-		panic(err)
-	}
-	// read connection message
-	msg := []PolgyonServerMsg{}
-	if err = c.ReadJSON(&msg); err != nil {
-		panic(err)
+	connectionAttempts := 0
+	for connectionAttempts < MaxConnectionAttempts {
+		connectionAttempts++
+		c, _, err := websocket.DefaultDialer.Dial(polygonStreamEndpoint, nil)
+		if err != nil {
+			if connectionAttempts == MaxConnectionAttempts {
+				panic(err)
+			}
+		} else {
+			// consume connection message
+			msg := []PolgyonServerMsg{}
+			if err = c.ReadJSON(&msg); err != nil {
```

Comment:
Mentioned this in Slack @ttt733 but this should actually be

```go
if err = c.ReadJSON(&msg); err == nil {
```

in order to continue!

Reply:
Yep - releasing a hotfix for that now.